src/fondant/executor.py

```diff
@@ -265,7 +277,16 @@ def _execute_component(
         Returns:
             A `dd.DataFrame` instance with initial data.
         """
-        return component.load()
+        dask_df = component.load()
+
+        if self.disable_automatic_indexing is False:
+            # Set monotonically increasing index
+            logger.info("Setting the index...")
+            dask_df["id"] = 1
+            dask_df["id"] = dask_df.id.cumsum()
+            dask_df = dask_df.reset_index(drop=True).set_index("id", sort=True)
```

Comment:
Is the `reset_index` necessary before `set_index`?

Reply:
I noticed that if an index is already set, using `set_index` will not change the initial index. I will check if there is a way to find out if an index is set

Reply:
I removed it, it was because I was setting the index twice in a graph (as part of a test) so it was only considering the first one